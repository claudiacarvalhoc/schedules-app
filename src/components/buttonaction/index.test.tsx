import React from 'react';
import ButtonAction from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const getShallowWrapper = ({ text, onClick = undefined }) => {
    return shallow(
        <ButtonAction text={text} onClick={onClick} />
    );
  };

describe('<Header />', () => {
    it('render correctly', () => {
        const props = {
            text: 'Retire',
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('handle correctly the onclick', () => {
        const mockOnRetire = jest.fn();
        const props = {
            text: 'Retire',
            onClick: mockOnRetire,
        };
        const component = getShallowWrapper(props);
        const button = component.first();
        button.simulate('click');
        expect(mockOnRetire).toHaveBeenCalled();
    });
});
