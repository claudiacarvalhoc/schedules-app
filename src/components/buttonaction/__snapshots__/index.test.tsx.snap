// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render correctly 1`] = `
<WithStyles(ForwardRef(Button))
  color="secondary"
  onClick={[Function]}
  size="small"
  variant="outlined"
>
  Retire
</WithStyles(ForwardRef(Button))>
`;
